src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -138,80 +181,137 @@ public Void call() throws Exception {
     @Override
     public String createDeviceMember(final String name, String ipAddress, String vserverIpAddress, String contactIpAddress,
             String gateway, String prefixLength) throws Exception {
-        return txControl.required(new Callable<DeviceMember>() {
+
+        return this.txControl.required(new Callable<DeviceMemberEntity>() {
+
             @Override
-            public DeviceMember call() throws Exception {
-                VSSDevice device = em.find(VSSDevice.class, vs.getId());
-                
-                DeviceMember dm = new DeviceMember();
-                dm.setParent(device);
-                dm.setName(name);
-                em.persist(dm);
-                return dm;
+            public DeviceMemberEntity call() throws Exception {
+
+                CriteriaBuilder criteriaBuilder = IsmDeviceApi.this.em.getCriteriaBuilder();
+                CriteriaQuery<DeviceEntity> query = criteriaBuilder.createQuery(DeviceEntity.class);
+                Root<DeviceEntity> r = query.from(DeviceEntity.class);
+                query.select(r)
+                        .where(criteriaBuilder.and(criteriaBuilder.equal(r.get("vsId"), IsmDeviceApi.this.vs.getId())));
+                List<DeviceEntity> result = IsmDeviceApi.this.em.createQuery(query).getResultList();
+
+                if (result.isEmpty()) {
+                    throw new Exception("Device Entity does not exists...");
+                    //TODO - to add RETURN 404 error:Sudhir
+                }
+                CriteriaBuilder memberBuilder = IsmDeviceApi.this.em.getCriteriaBuilder();
+                CriteriaQuery<DeviceMemberEntity> memberQuery = memberBuilder.createQuery(DeviceMemberEntity.class);
+                Root<DeviceMemberEntity> res = memberQuery.from(DeviceMemberEntity.class);
+                memberQuery.select(res).where(memberBuilder.and(memberBuilder.equal(r.get("name"), name)));
+
+                List<DeviceMemberEntity> deviceMemberResult = IsmDeviceApi.this.em.createQuery(memberQuery)
+                        .getResultList();
+                if (!deviceMemberResult.isEmpty()) {
+                    throw new Exception("Device Member Entity name already exists...:");
+                    //TODO - to add RETURN 404 error:Sudhir
```

Comment:
Same comment here, also look in other places

Reply:
OK